src/Switch/Switch.tsx

```diff
@@ -0,0 +1,147 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import {CloseIcon} from "../CloseIcon/CloseIcon";
+import Checkmark from "../Checkbox/CheckMark";
+
+import "./Switch.less";
+import {FlexBox, Justify} from "../flex";
+
+const propTypes = {
+  ariaLabelledby: PropTypes.string,
+  ariaLabel: PropTypes.string,
+  className: PropTypes.string,
+  checked: PropTypes.bool.isRequired,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  ariaLabelledby: "",
+  ariaLabel: "",
+  className: "",
+  disabled: false,
+};
+
+const cssClass = {
+  BG:  "Switch--bg",
+  CHECKED: "Switch--checked",
+  CHECKED_ICON: "Switch--checkedIcon",
+  CONTAINER: "Switch",
+  DISABLED: "Switch--disabled",
+  HANDLE: "Switch--handle",
+  HANDLE_GRIP_LEFT: "Switch--handleGripLeft",
+  HANDLE_GRIP_RIGHT: "Switch--handleGripRight",
+  UNCHECKED_ICON: "Switch--uncheckedIcon",
+};
+
+/**
+ * Switch toggles a single setting on or off with an immediate effect.
+ * Common uses:
+ *   - Enable or disable preferences, settings, or features.
+ *
+ * W3C spec: https://www.w3.org/TR/wai-aria-practices-1.1/#checkbox
+ */
+export default class Switch extends React.PureComponent {
+  static propTypes = propTypes;
+  static cssClass = cssClass;
+  static defaultProps = defaultProps;
+
+  state = {
+    hasOutline: false,
+    hover: false,
+    mouseDown: false,
+  };
+
+  render() {
+    const {
+      ariaLabelledby,
+      ariaLabel,
+      className,
+      checked,
+      disabled,
+    } = this.props;
+
+    const {hasOutline, hover} = this.state;
+
+    let bgStyle: any = {};
+
+    if (hasOutline) {
+      bgStyle = {
+        boxShadow: "0 0 0 6px rgba(128, 157, 255, 0.5)",
+      };
+    }
+    if (hover && !disabled) {
+      if (checked) {
+        bgStyle = {
+          ...bgStyle,
+          border: "2px solid #2DBE8B",
+          borderRadius: "4px",
+        };
+      } else {
+        bgStyle = {
+          ...bgStyle,
+          border: "2px solid #474C5E",
+          borderRadius: "4px",
+        };
+      }
+    }
+
+    return (
+      <div className={classnames(cssClass.CONTAINER, className)}>
+        <FlexBox
+          className={classnames(cssClass.BG, checked && cssClass.CHECKED, disabled && cssClass.DISABLED)}
+          style={bgStyle}
+          onMouseEnter={() => this.setState({hover: true})}
+          onMouseLeave={() => this.setState({hover: false})}
+          onMouseDown={() => this._onMouseDown()}
+          onMouseUp={() => this._onMouseUp()}
+        >
+          <FlexBox justify={Justify.BETWEEN} grow>
+            <FlexBox column justify={Justify.CENTER}><Checkmark className={cssClass.CHECKED_ICON} /></FlexBox>
+            <FlexBox column justify={Justify.CENTER}><CloseIcon className={cssClass.UNCHECKED_ICON} /></FlexBox>
+          </FlexBox>
+        </FlexBox>
+        <div
+          aria-checked={checked}
```

Comment:
these `aria` tags should also go on the top-level element, since that will be the clickable element